aptos-move/framework/src/module_metadata.rs

```diff
@@ -168,17 +192,47 @@ pub fn get_module_metadata(module: &CompiledModule) -> Option<RuntimeModuleMetad
     }
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, Error)]
+pub enum MetaDataValidationError {
+    #[error(transparent)]
+    Malformed(MalformedError),
+    #[error(transparent)]
+    InvalidAttribute(AttributeValidationError),
+}
+
+impl From<MalformedError> for MetaDataValidationError {
+    fn from(value: MalformedError) -> Self {
+        MetaDataValidationError::Malformed(value)
+    }
+}
+
+impl From<AttributeValidationError> for MetaDataValidationError {
+    fn from(value: AttributeValidationError) -> Self {
+        MetaDataValidationError::InvalidAttribute(value)
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Error)]
+pub enum MalformedError {
+    #[error("Unknown key found: {0:?}")]
+    UnknownKey(Vec<u8>),
+    #[error("Unable to deserialize value for {0:?}: {1}")]
+    DeserializedError(Vec<u8>, bcs::Error),
+    #[error("Found more than one metadata entries: {0}")]
+    MultipleEntries(u64),
+}
+
 #[derive(Clone, Debug, PartialEq, Eq, Error)]
 #[error("Unknown attribute ({}) for key: {}", self.attribute, self.key)]
-pub struct MetadataValidationError {
+pub struct AttributeValidationError {
```

Comment:
Since the name is not used in serde, this is a safe backward-compatible change right?

Reply:
I believe the error is not persisted but just used for logging